components/brave_wallet/browser/internal/hd_key_utils.cc

```diff
@@ -0,0 +1,65 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/internal/hd_key_utils.h"
+
+#include <optional>
+
+#include "base/containers/span.h"
+#include "base/numerics/checked_math.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/string_split.h"
+#include "crypto/process_bound_string.h"
+
+namespace brave_wallet {
+
+namespace internal {
+void SecureZeroBuffer(base::span<uint8_t> data) {
+  crypto::internal::SecureZeroBuffer(data);
+}
+
+}  // namespace internal
+
+std::optional<std::vector<uint32_t>> ParseFullHDPath(std::string_view path) {
+  auto entries = base::SplitStringPiece(
+      path, "/", base::WhitespaceHandling::TRIM_WHITESPACE,
+      base::SplitResult::SPLIT_WANT_ALL);
+
+  if (entries.empty()) {
+    return std::nullopt;
+  }
+  if (entries.front() != kMasterNode) {
+    return std::nullopt;
+  }
+
+  std::vector<uint32_t> result;
+  result.reserve(entries.size() - 1);
+  for (auto node : base::span(entries).subspan(1)) {
+    uint32_t offset = 0;
+    if (node.ends_with('\'')) {
```

Comment:
please add comments for the expected format(s)

Reply:
removed completely